### ISSUE TITLE
Scrolling or zooming not updating CTM

### ISSUE THREAD

**Issue by dgroh** [ariutta/svg-pan-zoom#314]
### Expected behaviour
Example **working** **without** `svg-pan-zoom`: http://jsfiddle.net/4wv07ntz/72/

To test, just drag the circles around and **use the scroll bar to scroll down a little bit** and drag the elements again.

### Actual behaviour
When I drag an circle around, everything works fine until I use the scrollbar and scroll down a little bit and try to drag the circle again, the cursor loses it's reference.

Example **not working** **with** `svg-pan-zoom`: http://jsfiddle.net/4zekbrLf/1/

### Steps to reproduce the behaviour

1. Opens the **not working** **with** `svg-pan-zoom` jsFiddle
2. Drag one of the circles around
3. Use the scrollbar to scroll down a little bit
4. Drag any of the elements again

![image](https://user-images.githubusercontent.com/16785357/45541169-93970100-b80e-11e8-8920-5769f75f8303.png)

### Configuration

- svg-pan-zoom version: ``
- Browser(s): ``
- Operating system(s): ``
- A relevant example URL:

Feature Request
---

- Feature description
- Reasons for adopting new feature
- Is this a breaking change? (How will this affect existing functionality)

**Comment by dgroh** [comment]
I could fix this by passing the offsets variables to `mousedown` instead of in `mousemove` which also makes the code cleaner, but I'm interested to know why this is working without `Svg-Pan-Zoom`

**Comment by bumbu** [comment]
Hi @dgroh. 
I should say that this is a really nice functionality that you implemented. People often ask for this functionality - and now we have a really clean example of how to do that. This is a great job!

Regarding the issue. I tried reproducing and issue and for me (macOS, Chrome 68) the bug doesn't happen. If I use the scrollbar - it works perfectly fine. 
I also added a console.log in `onUpdatedCTM` to make sure if this is an issue of CTM not being updated - but that also works fine. I can see the CTM being updated with what seems correct values on pan and zoom.

After fiddling a bit more with the demo I managed to understand how to reproduce the bug:
* If you first scroll/zoom/pan then try to drag the dot - it works fine
* If you drag a dot, then scroll/zoom/pan then try to drag again - there's an offset. If you get back to pre-scroll/pre-zoom/pre-pan position - it works ok

That seemed weird to me and after looking through code I found where the offset is set: `if (!byPassOffset) {...`. And from described behaviour is seemed that the offset is set only once. Checking the code - and that turned out to be the case. 

Here's the [working example](http://jsfiddle.net/bumbu/4zekbrLf/6/). All it needed is to set the `byPassOffset` to false, so that at next drag - the offset is going to be recalculated again. 

And that's why moving the setting of offset to `mousedown` and getting rit of `byPassOffset` would work - you'd do it at the beginning of each drag (which is correct).
